### PR TITLE
hashFirstRowComponent for Indices

### LINKED ISSUE #980: Should be able to set hashFirstRowComponent on index definitions
Can only currently set it on tables; but don't see why it shouldn't be viable for indices as well.

### PR DESCRIPTION
**Pull request by j-baker**
Fixes #980.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/986)

### CODE REVIEW

File: docs/source/release_notes/release-notes.rst [hunk 1]
```diff
@@ -31,6 +31,22 @@ Changelog
...
+
+.. list-table::
+    :widths: 5 40
+    :header-rows: 1
+
+    *    - Type
+         - Change
+
+    *    - |new|
+         - ``hashFirstRowComponent`` can now be used on index definitions as well.
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: We also add a link to the pull request in the release notes.

**Comment by gsheasby** [comment]
Looks good. Would like to see a unit test or two, and documentation. Do you have the bandwidth?

File: atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/IndexDefinition.java [hunk 2]
```diff
@@ -92,6 +92,20 @@ public void componentFromDynamicColumn(String componentName, ValueType valueType
     public void componentFromDynamicColumn(String componentName, ValueType valueType, ValueByteOrder valueByteOrder, String sourceComponentName) {
         addComponent(IndexComponent.createFromDynamicColumn(new NameComponentDescription(componentName, valueType, valueByteOrder), sourceComponentName));
     }
+    /**
```
Comment (schlosna) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/IndexDefinition.java]: nit: blank line before JavaDoc

**Comment by j-baker** [comment]
Any chance an Atlassian could pair with me today on writing tests? Whilst I would normally properly unit test, there are no tests for this code path, so it would be good to work out something that would be acceptable.

**Comment by gsheasby** [comment]
We have all of the meetings today, but someone (possibly me) can help you out tomorrow.

**Comment by schlosna** [comment]
👍  once circle passes, previous build died of dysentery `./scripts/circle-ci/run-circle-tests.sh died unexpectedly`, so I just kicked it

**Comment by gsheasby** [comment]
Reviewed 1 of 3 files at r1, 6 of 6 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/986)*

**Pull request merged**